src/main/charts/bamboo/values.yaml

```diff
@@ -609,6 +613,46 @@ bamboo:
     #
     failureThreshold: 30
 
+    # -- Custom readinessProbe to override the default /status httpGet
+    #
+    customProbe: {}
+#      tcpSocket:
+#        port: 8080
+#      periodSeconds: 5
+#      failureThreshold: 120
+
+  # Confirm that Bamboo is up and running with a StartupProbe
+  # https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes
+  #
+  startupProbe:
+
+    # -- How often (in seconds) the Bamboo container startup probe will run
+    #
+    periodSeconds: 5
+
+    # -- The number of consecutive failures of the Bamboo container startup probe
+    # before the pod fails startup checks.
+    #
+    failureThreshold: 120
+
+  # Ensure that the server responds with a LivenessProbe
+  # https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request
+  #
+  livenessProbe:
+
+    # -- Whether to apply the livenessProbe check to pod.
+    #
+    enabled: true
```

Comment:
What led us to introduce the Liveness Probe? When we were working on the helm charts, I remember we didn't introduce it because it can end up in a situation when the product is in a restart loop when it becomes unresponsive or slow. When we will add it `enabled: true`  as default, can it potentially affect existing customers that will upgrade and start experiencing new behaviour?

Reply:
LivenessProbe for Jira was introduced in https://github.com/atlassian/data-center-helm-charts/pull/511

Yes, if readinessProbe (tcpSocket) fails, the pod will be restarted, and yes, kubelet will restart it every time it fails. The difference with the readinessProbe is that it's tcpSocket check, not httpGet /status. I think it's fair for kubelet to restart a pod if it cannot open a socket connection to the server port in the pod (which means the application is unresponsive).

Yes, with the default value of true, statefulsets will be updated and pods will restart after helm upgrade. Do you think false is a safer and more conservative/non-disruptive option?

Reply:
Is it possible that the server won't open or being responsive on the port at all for some time?

To be frank, I am not sure if we need to stick to the `false` to the default for Liveness Probe. I am just worried about potential unknown implications for large enterprise instances that we don't have testing or exposure of.

Reply:
> Is it possible that the server won't open or being responsive on the port at all for some time?
that's a possibility, yes. Perhaps, we can make the default `failureThreshold` a bit higher, say, 12? This way, the pod will be restarted if kubelet is unable to open a socket on the server port for 1 minute.

Reply:
I think that does make sense - we should have some more lenient default values to potentially don't have regressive impact on existing customer instances. After a bit of :chatting: I am convinced that it might take some time to open the Tomcat connector to make the socket accessible.

> Socket is accessible in Tomcat after it starts. When Tomcat starts up, it initializes the connectors for the specified ports and starts listening for incoming connections on those ports. Once Tomcat has completed its startup process, including loading the web applications and initializing the servlet contexts, the socket is accessible and ready to accept incoming connections.

> Keep in mind that the actual time for the socket to become accessible may vary depending on factors such as the size of your application, the number of web applications being deployed, and the resources available on your system. To ensure that the liveness probes in Kubernetes do not start checking the health of your application before the socket is accessible, you can configure the `initialDelaySeconds` parameter in the liveness probe configuration. This will give your Tomcat Java application enough time to start and initialize the socket before Kubernetes starts performing the health checks.